Library/Source/NativeEngine.cpp

```diff
@@ -1011,9 +1006,52 @@ namespace Babylon
             static_cast<bgfx::TextureFormat::Enum>(image.m_format),
             0,
             imageDataRef);
+    }
+
+    Napi::Value NativeEngine::LoadTexture(const Napi::CallbackInfo& info)
+    {
+        const auto textureData = info[0].As<Napi::External<TextureData>>().Data();
+        const auto buffer = info[1].As<Napi::ArrayBuffer>();
+        const auto mipMap = info[2].As<Napi::Boolean>().Value();
+        const auto invertY = info[3].As<Napi::Boolean>().Value();
+
+        textureData->Images.push_back(bimg::imageParse(&m_allocator, buffer.Data(), static_cast<uint32_t>(buffer.ByteLength())));
+        auto& image = *textureData->Images.front();
+
+        ImageToTexture(textureData, image, invertY, mipMap);
         return Napi::Value::From(info.Env(), bgfx::isValid(textureData->Texture));
     }
 
+    void NativeEngine::UpdateRawTexture(const Napi::CallbackInfo& info)
+    {
+        const auto textureData = info[0].As<Napi::External<TextureData>>().Data();
+        const auto width = static_cast<uint32_t>(info[2].As<Napi::Number>().Uint32Value());
+        const auto height = static_cast<uint32_t>(info[3].As<Napi::Number>().Uint32Value());
+        const auto formatIndex = info[4].As<Napi::Number>().Uint32Value();
+        const auto mipMap = info[5].As<Napi::Boolean>().Value();
+        const auto invertY = info[6].As<Napi::Boolean>().Value();
+
+        const void* data = nullptr;
+
+        switch(formatIndex)
```

Comment:
nit: probably should format the whole document?

Reply:
I can do that in a separate PR.